### ISSUE TITLE
run demo generation failed

### ISSUE THREAD

**Issue by biaochen** [triton-inference-server/tensorrtllm_backend#316]
### System Info

x86_64
V100
triton server image: nvcr.io/nvidia/tritonserver:23.12-trtllm-python-py3
tensorrtllm_backend: v0.7.1

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

1. modify model repos as document said;
2. docker run nvcr.io/nvidia/tritonserver:23.12-trtllm-python-py3,
3. git clone tensorrtllm_backend with v0.7.1 tag
4. start triton server by running `python3 scripts/launch_triton_server.py --model_repo=xxx --force`
log shows that all models haved been loaded successfully:
+----------------+---------+--------+
| Model          | Version | Status |
+----------------+---------+--------+
| ensemble       | 1       | READY  |
| postprocessing | 1       | READY  |
| preprocessing  | 1       | READY  |
| tensorrt_llm   | 1       | READY  |

5. run generation:
curl -X POST localhost:8000/v2/models/ensemble/generate -d '{"text_input": "What is machine learning?", "max_tokens": 20, "bad_words": "", "stop_words": ""}'
got following response:
{"error":"in ensemble 'ensemble', Failed to process the request(s) for model instance 'preprocessing_0_0', message: AttributeError: 'int' object has no attribute 'decode'\n\nAt:\n  /test/trtllm_repo/preprocessing/1/model.py(217): <listcomp>\n  /test/trtllm_repo/preprocessing/1/model.py(214): _create_request\n  /test/trtllm_repo/preprocessing/1/model.py(169): execute\n"}

### Expected behavior

generation api succeed

### actual behavior

got following response:
{"error":"in ensemble 'ensemble', Failed to process the request(s) for model instance 'preprocessing_0_0', message: AttributeError: 'int' object has no attribute 'decode'\n\nAt:\n  /test/trtllm_repo/preprocessing/1/model.py(217): <listcomp>\n  /test/trtllm_repo/preprocessing/1/model.py(214): _create_request\n  /test/trtllm_repo/preprocessing/1/model.py(169): execute\n"}

### additional notes

no

**Comment by biaochen** [comment]
guess maybe configuration issue? should I upload configuration files;

**Comment by byshiue** [comment]
Could you try print the related variables in https://github.com/triton-inference-server/tensorrtllm_backend/blob/v0.7.1/all_models/inflight_batcher_llm/preprocessing/1/model.py#L210?

**Comment by mse700** [comment]
I have the same issue with the llama2 deployment using TensoRT Backend. 

Is there any timeline about fixing this bug?

**Comment by byshiue** [comment]
Because we cannot reproduce your issue, so we cannot provide the timeline about fixing. 

Also, might you try the latest main branch because there are many updates after v0.7.1.

**Comment by mse700** [comment]
I have the exact Error which is reported above. In fact, I have clone the latest version from main branch and followed the described steps in this repository. it seems that there is an error in preprocessing model.py :

{"error":"in ensemble 'ensemble', Failed to process the request(s) for model instance 'preprocessing_0_0', message: AttributeError: 'int' object has no attribute 'decode'\n\nAt:\n /test/trtllm_repo/preprocessing/1/model.py(217): \n /test/trtllm_repo/preprocessing/1/model.py(214): _create_request\n /test/trtllm_repo/preprocessing/1/model.py(169): execute\n"}

which causes the model not to perform and break down the server start.
Reproduction

    modify model repos as document said;

    docker run nvcr.io/nvidia/tritonserver:23.12-trtllm-python-py3,

    git clone tensorrtllm_backend with v0.7.1 tag

    start triton server by running python3 scripts/launch_triton_server.py --model_repo=xxx --force
    log shows that all models haved been loaded successfully:
    +----------------+---------+--------+
    | Model | Version | Status |
    +----------------+---------+--------+
    | ensemble | 1 | READY |
    | postprocessing | 1 | READY |
    | preprocessing | 1 | READY |
    | tensorrt_llm | 1 | READY |

    run generation:
    curl -X POST localhost:8000/v2/models/ensemble/generate -d '{"text_input": "What is machine learning?", "max_tokens": 20, "bad_words": "", "stop_words": ""}'
    got following response:
    {"error":"in ensemble 'ensemble', Failed to process the request(s) for model instance 'preprocessing_0_0', message: AttributeError: 'int' object has no attribute 'decode'\n\nAt:\n /test/trtllm_repo/preprocessing/1/model.py(217): \n /test/trtllm_repo/preprocessing/1/model.py(214): _create_request\n /test/trtllm_repo/preprocessing/1/model.py(169): execute\n"}

**Comment by byshiue** [comment]
Could you share the full reproduced steps instead of only sharing the scripts of launching server? 

Also, please check again that you really use the latest main branch. For example, in `all_models/inflight_batcher_llm/preprocessing/1/model.py` of latest main branch, the [line 214](https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/all_models/inflight_batcher_llm/preprocessing/1/model.py#L214) is comments. It does not match your error.

**Comment by mse700** [comment]
Sorry for the late respond.
I have realized the problem. I am deploying the model on 2 H100 80GB NVIDA GPUs with the `nvcr.io/nvidia/tritonserver:24.03-trtllm-python-py3` image. As of my understanding of the image and my google searchs, I can not set the `world_size` directly with the image entry_point (tritonserver), therefor I needed to run the python code [here](https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/scripts/launch_triton_server.py) in order to set the world_size parameter as input parameter and then launch the triton inference server.
